packages/api/models/collections.js

```diff
@@ -64,6 +65,41 @@ class Collection extends Manager {
 
     return super.create(item);
   }
+
+  /**
+   * Delete a collection
+   *
+   * @param {string} name - the collection name
+   * @param {string} version - the collection version
+   */
+  async delete(name, version) {
+    if (!(await this.exists(name, version))) throw new Error('Collection does not exist');
+
+    if (await this.hasAssociatedRules(name, version)) {
+      throw new Error('Cannot delete a collection that has associated rules');
```

Comment:
Do you think there's any value in surfacing the associated rules in the error response from the endpoint?